### DISCUSSION TITLE
Unable to get the ui working with any version of the helm chart deployment

### DISCUSSION THREAD

**Discussion by rinnegan72** [openbao/openbao#799] [Q&A] [answered]
To my understanding open bao is basically just hashicorp vault but open souce but there is a problem for me at least. when I install vault, do init and unseal with the vault operator vault starts working and I am able to use the dashboard with the root token. Taking a look at the docs bao seemed no different so I tried the same thing but I keep seeing the ui unavaiable page.  I thought maybe its a values. issue and even with ui enabled true it does not work and while I now see the a seperate ui service I am still unable to access the dashboard.  doing this all on an m1 macbook air if somehow the issue is that bao does not support arm 64
<img width="665" alt="image" src="https://github.com/user-attachments/assets/243f1ade-58be-4694-92a8-80fc9860f2f2" />

```
ui:
  enabled: true
```
I tried using this standalone config provided in the values file for the chart's default values:
```
server:
  standalone:
   config: |
      ui = true

      listener "tcp" {
        tls_disable = 1
        address = "[::]:8200"
        cluster_address = "[::]:8201"
        # Enable unauthenticated metrics access (necessary for Prometheus Operator)
        #telemetry {
        #  unauthenticated_metrics_access = "true"
        #}
      }
      storage "file" {
        path = "/openbao/data"
      }

      # Example configuration for using auto-unseal, using Google Cloud KMS. The
      # GKMS keys must already exist, and the cluster must have a service account
      # that is authorized to access GCP KMS.
      #seal "gcpckms" {
      #   project     = "openbao-helm-dev"
      #   region      = "global"
      #   key_ring    = "openbao-helm-unseal-kr"
      #   crypto_key  = "openbao-helm-unseal-key"
      #}

      # Example configuration for enabling Prometheus metrics in your config.
      #telemetry {
      #  prometheus_retention_time = "30s"
      #  disable_hostname = true
      #}
ui:
  enabled: true
 ```
It did not work please help

**Comment by cipherboy** [comment, marked as answer]
\o hey @rinnegan72 -- see the [notes on GitHub releases](https://github.com/openbao/openbao/releases/tag/v2.1.0) and on our [release notes](https://openbao.org/docs/release-notes/2-1-0/) --- the UI was difficult to fork without experience on it and so we opted to avoid forking it for the time being. It is expected that the UI is not present. 

https://github.com/openbao/openbao/issues/739 tracks the reintroduction of the UI; until then, you'd have to compile OpenBao (and UI) from scratch.

**Reply by rinnegan72** [reply]
Please mention this in the documentation, the documentation is basically a 1 for 1 of vault with vault replaced with openbao when it does not have this in the helm chart unless we compile it from source.
<img width="835" alt="image" src="https://github.com/user-attachments/assets/15d8b0aa-6215-482c-8b57-ab6d746ca28d" />
<img width="746" alt="image" src="https://github.com/user-attachments/assets/2f75bd59-b61c-4e16-8bc5-58d8c12bc619" />

**Reply by cipherboy** [reply]
> Please mention this in the documentation, the documentation is basically a 1 for 1 of vault with vault replaced with openbao when it does not have this in the helm chart unless we compile it from source.

This is because OpenBao is a fork of Vault and the documentation was in-repository and licensed permissively at the time of the fork. :-) 

Many users are compiling the UI from source while it has been under redevelopment; it is a viable approach for the helm chart as well if you use a replacement container image. 

That said, if you'd like to add a warning that this section of the docs is under development, happy to take a PR.

**Reply by rinnegan72** [reply]
> > Please mention this in the documentation, the documentation is basically a 1 for 1 of vault with vault replaced with openbao when it does not have this in the helm chart unless we compile it from source.
> 
> This is because OpenBao is a fork of Vault and the documentation was in-repository and licensed permissively at the time of the fork. :-)
> 
> Many users are compiling the UI from source while it has been under redevelopment; it is a viable approach for the helm chart as well if you use a replacement container image.
> 
> That said, if you'd like to add a warning that this section of the docs is under development, happy to take a PR.

For sure it's cool that we have to build the ui from source. The idea being that if we want the ui now, we build the required custom image and then use that custom image with the ui but I think the documentation should at least make it clear that it is expected otherwise it feels more like I did something wrong and should try again you get me? I did create a pr with a warning hopefully this looks good to you https://github.com/openbao/openbao/pull/803
